src/plugins/intel_cpu/src/emitters/snippets/x64/kernel_executors/brgemm_copy_b.cpp

```diff
@@ -0,0 +1,316 @@
+      wei_data_size(dnnl_data_type_size(conf.get_wei_dt())), vnni_factor(data_type_vnni_granularity(conf.get_wei_dt())),
+      K(conf.get_K()), N_blk(conf.get_N_blk()), wei_N_blk(conf.get_wei_N_blk()), wei_N_tail(conf.get_wei_N_tail()), ker_(nullptr) {
+    init_brgemm_copy_b_kernel(dnnl_brgemm_copy_b_kernel, conf);
+    OV_CPU_JIT_EMITTER_ASSERT(dnnl_brgemm_copy_b_kernel, "Kernel is missed!");
+}
+
+status_t BrgemmCopyBKernel::create_kernel() {
+    const auto code = jit_generator::create_kernel();
+    OV_CPU_JIT_EMITTER_ASSERT(code == status::success, "Failed to create kernel");
+    ker_ = (decltype(ker_))jit_ker();
+    return code;
+}
+
+void BrgemmCopyBKernel::operator()(const call_args* args) const {
+    OV_CPU_JIT_EMITTER_ASSERT(ker_, "Kernel is nullptr");
+    ker_(args);
+}
+
+void BrgemmCopyBKernel::init_brgemm_copy_b_kernel(std::unique_ptr<dnnl::impl::cpu::x64::matmul::jit_brgemm_matmul_copy_b_t>& kernel,
+                                                  const BrgemmCopyBKernelConfig& conf) const {
+    matmul::brgemm_matmul_conf_t brgCopyKernelConf;
+    brgCopyKernelConf.src_dt = conf.get_src_dt();
+    brgCopyKernelConf.wei_dt = conf.get_wei_dt();
+    brgCopyKernelConf.orig_wei_dt = brgCopyKernelConf.wei_dt;
+    brgCopyKernelConf.wei_n_blk = static_cast<int>(conf.get_wei_N_blk());
+    // Note: 2D format tags are used just to force the needed OneDNN primitive creation.
+    // However, the generated primitive can be also applied to tensors with other ranks
+    brgCopyKernelConf.wei_tag = conf.is_transposed_B() ? dnnl_ba : dnnl_ab;
+    brgCopyKernelConf.transposed_B = conf.is_transposed_B();
+    brgCopyKernelConf.copy_B_wei_stride = conf.get_copy_B_wei_stride();
+    brgCopyKernelConf.LDB = conf.get_LDB();
+    brgCopyKernelConf.N = conf.get_N();
+    brgCopyKernelConf.N_tail = conf.get_wei_N_tail();
+    brgCopyKernelConf.N_blk = conf.get_wei_N_blk();
+    brgCopyKernelConf.K = conf.get_K_blk();
+    brgCopyKernelConf.K_blk = conf.get_K_blk();
+    brgCopyKernelConf.N_chunk_elems = brgCopyKernelConf.N_blk;
+    brgCopyKernelConf.b_dt_sz = DnnlExtensionUtils::sizeOfDataType(static_cast<dnnl::memory::data_type>(brgCopyKernelConf.wei_dt));
+    brgCopyKernelConf.tr_b_dt_sz = DnnlExtensionUtils::sizeOfDataType(static_cast<dnnl::memory::data_type>(brgCopyKernelConf.wei_dt));
+
+    brgCopyKernelConf.req_wei_vnni_downconvert = false;
+
+    brgCopyKernelConf.isa = conf.get_isa();
+    brgCopyKernelConf.s8s8_compensation_required = conf.is_with_comp();
+
+    brgCopyKernelConf.has_zero_point_a = false;
+    brgCopyKernelConf.has_zero_point_b = false;
+    brgCopyKernelConf.src_zp_type = dnnl::impl::cpu::x64::none;
+
+    OV_CPU_JIT_EMITTER_ASSERT(matmul::create_brgemm_matmul_copy_b(kernel, &brgCopyKernelConf) == dnnl_success,
+                              "cannot create kernel due to invalid params");
+}
+
+void BrgemmCopyBKernel::generate() {
```

Comment:
JIT inside a JIT - cool 😎 
BTW do you think we gain any performance here due to jitting? If we just need to call tail + block kernels ad do some pointer increments, probably we can do in just in C++? Do you think we can use similar c++ wrappers in future? It'll be just easier to support and debug.